src/backend/catalog/index.c

```diff
@@ -750,8 +750,10 @@ index_create(Relation heapRelation,
  		 * Index of Table Variables should also be in ENR and the index
 		 * can only be created as part of DECLARE @[name] TABLE;
 		 * when specifying PRIMARY KEYs and/or CONSTRAINTs.
+		 *
+		 * Indexes of temp tables should be in enr as well, to prevent cache issues.
 		 */
-		is_enr = (indexRelationName && strlen(indexRelationName) > 0 && indexRelationName[0] == '@');
+		is_enr = (indexRelationName && strlen(indexRelationName) > 0 && (indexRelationName[0] == '@' || indexRelationName[0] == '#'));
```

Comment:
In `heap_create_with_catalog()` you have a check there so not all temp tables are added in ENR but here you are doing all "index" of temp tables should be in enr. What happens in the case where the temp table is NOT in enr but index is in enr?

Reply:
In that case, this line might be unnecessary.

Reply:
we need to follow all in ENR or nothing. We should put indexes in ENR if corresponding tables are in ENR. We should have same check as that for tables.

Reply:
agreed, reworked to handle indexes in ENR using same check from heap_create_with_catalog.